src/types/content.rs

```diff
@@ -0,0 +1,176 @@
+use std::collections::HashMap;
+use ::types::messages::{ImageInfo};
+
+#[derive(Serialize, Deserialize, Debug)]
+#[serde(rename_all="snake_case")]
+/// Information about whether people are online or not.
+pub enum Presence {
+    Online,
+    Offline,
+    Unavailable
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+#[serde(rename_all="snake_case")]
+/// Defines who can join a room
+pub enum JoinRule {
+    Public,
+    Invite,
+    // reserved keywords
+    // Knock,
+    // Private,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+#[serde(rename_all="snake_case")]
+/// Defines what Membership a user in a room
+pub enum Membership {
+    Invite,
+    Join,
+    Leave,
+    Ban,
+    // reserved word
+    // Knock,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+#[serde(rename_all="snake_case")]
+/// Defines what Membership a user in a room
```

Comment:
I think you got a bit copy-pasta happy here... either remove, or replace with something that makes sense :P